tests/dotnet/UnitTests/XcodeProjectTests.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+
+using Microsoft.Build.Logging.StructuredLogger;
+
+namespace Xamarin.Tests {
+	public class XcodeProjectTests : TestBaseClass {
```

Comment:
@rolfbjarne Before I go too far down this road, is this the preferred place to add new MSBuild tests?

Reply:
Yes, this works for tests that involve complete projects.

For testing single task or targets, there are test suites here:

> https://github.com/xamarin/xamarin-macios/tree/main/tests/msbuild